edgelet/edgelet-docker/src/runtime.rs

```diff
@@ -1272,13 +1276,53 @@ fn get_notary_parameters(
     Some((notary_auth, gun, tag, config_path.to_path_buf()))
 }
 
+// Disallow adding privileged and other capabilities if allow_privileged is false
+fn unset_privileged(allow_privileged: bool, create_options: &mut ContainerCreateBody) {
+    if let Some(config) = create_options.host_config() {
+        if !allow_privileged && (config.privileged() == Some(&true) || !config.cap_add().is_empty())
+        {
+            warn!("Privileged capabilities are disallowed on this device. Privileged capabilities can be used to gain root access. If a module needs to run as privileged, and you are aware of the consequences, set `allow_privileged` to `true` in the config.toml and restart the service.");
+            let mut config = config.clone();
+
+            config.set_privileged(false);
+            config.reset_cap_add();
+
+            create_options.set_host_config(config);
+        }
+    }
+}
+
+fn drop_unsafe_privileges(create_options: &mut ContainerCreateBody) {
+    // These capabilities are provided by default and can be used to gain root access:
+    // https://labs.f-secure.com/blog/helping-root-out-of-the-container/
+    // They must be explicitly enabled
+    let mut caps_to_drop = vec!["CAP_CHOWN".to_owned(), "CAP_SETUID".to_owned()];
+
+    // The suggested `Option::map_or_else` requires cloning `caps_to_drop`.
+    #[allow(clippy::option_if_let_else)]
+    let host_config = if let Some(config) = create_options.host_config() {
+        // Don't drop if customer specifies explicitly
+        caps_to_drop.retain(|cap_drop| !config.cap_add().contains(cap_drop));
+
+        // Add customer specified cap_drops
+        caps_to_drop.extend_from_slice(config.cap_drop());
+
+        config.clone().with_cap_drop(caps_to_drop)
```

Comment:
Same as above, does `if let Some(mut config)` on 1303 work? If so, you can remove the clone.